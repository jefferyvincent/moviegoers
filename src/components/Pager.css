.pager {
    width: 100%;
    margin-top: 1rem;
    text-align: center;
}

.btn {
    display: inline-block;
    width: 140px;
    height: 59px;
    cursor: pointer;
    background-color: #00c8ff;
    color: white;
    line-height: 3rem;
    position: relative;
}

.prev-btn {
    margin-right: 1rem;
}

.prev-btn::after {
    content: "";
    position: absolute;
    left: -30px;
    width: 0;
    height: 0;
    border-top: 30px solid transparent;
    border-bottom: 30px solid transparent;
    border-right: 30px solid #00c8ff;
}

.next-btn:after {
    content: "";
    position: absolute;
    right: -30px;
    width: 0;
    height: 0;
    border-top: 30px solid transparent;
    border-bottom: 30px solid transparent;
    border-left: 30px solid #00c8ff;
}

.disabled {
    display: none;
}

@media only screen and (max-width: 904px) {
    .btn {
        width: 100px;
        margin: 1rem;
    }
}