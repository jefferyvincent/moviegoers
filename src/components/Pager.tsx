import React, {useContext} from 'react';

import Context from '../Context';
import './Pager.css';

interface PagerProps {
    page: number;
    max: number;
    history: any;
}

const Pager: React.FC<PagerProps> = ({page, max, history}) => {
    const dispatch:any = useContext(Context);

    const scrollTop = ():void => {
        document.body.scrollTop = 0; // For Safari
        document.documentElement.scrollTop = 0; // For Chrome, Firefox, IE and Opera
    }

    const pageNext = ():void => {
        if (page <= max ) {
            
            page++

            dispatch({
                type: 'updatePage',
                payload: page 
            });

            scrollTop();
            history.push('/movies/'+page);
        }
    }

    const pagePrev = ():void => {
        if (page >= 1) {
            page--;

            dispatch({
                type: 'updatePage',
                payload: page
            });

            scrollTop();
            history.push('/movies/'+page);
        }
    }

    let prevClass = 'prev-btn btn';
    let nextClass = 'next-btn btn';

    if (page <= 1) {
        prevClass = prevClass + " disabled";
    }

    if (page >= max) {
        nextClass = nextClass + " disabled";
    }

    if (max > 0) {
        return (
            <div className="pager">
                <span className={prevClass} onClick={()=>{pagePrev()}}>
                    Previous
                </span>
                <span className={nextClass} onClick={()=>{pageNext()}}>
                    Next
                </span>
            </div>
        )
    } else {
        return null;
    }
}

export default Pager;