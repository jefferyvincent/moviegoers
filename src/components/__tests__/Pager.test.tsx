import React from 'react';
import { shallow } from 'enzyme';
import {render } from 'react-testing-library';
import {HashRouter as Router} from "react-router-dom";

import Pager from '../Pager';

describe('<Pager/>', () => {
  let wrapper:any;
  beforeEach(() => { wrapper = shallow(<Pager page={1} max={10} history={[]} />); });

  it("renders", () => {
    const { asFragment } = render(<Router><Pager page={1} max={10} history={[]} /></Router>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders a div', () => {
    expect(wrapper.find('div.pager').length).toEqual(1);
  });
});