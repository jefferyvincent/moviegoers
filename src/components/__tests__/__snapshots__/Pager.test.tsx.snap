// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pager/> renders 1`] = `
<DocumentFragment>
  <div
    class="pager"
  >
    <span
      class="prev-btn btn disabled"
    >
      Previous
    </span>
    <span
      class="next-btn btn"
    >
      Next
    </span>
  </div>
</DocumentFragment>
`;
